tabrepo/repository/ensemble_mixin.py

```diff
@@ -1,83 +1,286 @@
+                rng = np.random.default_rng(seed=seed)
+                configs_fit_order = list(rng.permuted(configs))
+            else:
+                configs_fit_order = configs
+
+            configs = filter_configs_by_runtime(repo=self, dataset=dataset, fold=fold, config_names=configs_fit_order, max_cumruntime=time_limit)
+
+            if len(configs) == 0:
+                if self._config_fallback is None:
+                    if len(configs_fit_order) > 0:
+                        raise AssertionError(
+                            f"Can't fit an ensemble with no configs when self._config_fallback is None "
+                            f"(No configs are trainable in the provided time_limit={time_limit}.)"
+                        )
+                    else:
+                        raise AssertionError(f"Can't fit an ensemble with no configs when self._config_fallback is None.")
+                configs = [self._config_fallback]
+
         scorer = self._construct_ensemble_selection_config_scorer(
-            tasks=tasks,
+            tasks=[task],
             ensemble_size=ensemble_size,
             ensemble_cls=ensemble_cls,
             ensemble_kwargs=ensemble_kwargs,
-            backend=backend,
+            backend="native",
         )
 
-        dict_errors, dict_ensemble_weights = scorer.compute_errors(configs=configs)
+        results = scorer.compute_errors(configs=configs)
+        metric_error = results[task]["metric_error"]
+        ensemble_weights = results[task]["ensemble_weights"]
+        metric_error_val = results[task]["metric_error_val"]
+
+        dataset_info = self.dataset_info(dataset=dataset)
+        metric = dataset_info["metric"]
+        problem_type = dataset_info["problem_type"]
+
+        # select configurations used in the ensemble as infer time only depends on the models with non-zero weight.
+        fail_if_missing = self._config_fallback is None
+        config_selected_ensemble = [
+            config for i, config in enumerate(configs) if ensemble_weights[i] != 0
+        ]
+
+        runtimes = get_runtime(
+            repo=self,
+            dataset=dataset,
+            fold=fold,
+            config_names=configs,
+            runtime_col='time_train_s',
+            fail_if_missing=fail_if_missing,
+        )
+        latencies = get_runtime(
+            repo=self,
+            dataset=dataset,
+            fold=fold,
+            config_names=config_selected_ensemble,
+            runtime_col='time_infer_s',
+            fail_if_missing=fail_if_missing,
+        )
+        time_train_s = sum(runtimes.values())
+        time_infer_s = sum(latencies.values())
+
+        output_dict = {
+            "metric_error": [metric_error],
+            "metric": [metric],
+            "time_train_s": [time_train_s],
+            "time_infer_s": [time_infer_s],
+            "problem_type": [problem_type],
+            "metric_error_val": [metric_error_val],
+        }
```

Comment:
If we put this code in the main class, I think it would be good to put high level comments on the key blocks to make the code more readable (could be a todo).

Reply:
added some inline comments